packages/extensions/modelerfour/src/quality-precheck/prechecker.ts

```diff
@@ -274,17 +274,29 @@ export class QualityPreChecker {
           }
 
           if (schema.allOf || schema.anyOf || schema.oneOf) {
-            // if the model has properties, then we're going to assume they meant to say JsonType.object
-            // but we're going to warn them anyway.
-            this.session.warning(
-              `The schema '${
-                schema?.["x-ms-metadata"]?.name || name
-              }' with an undefined type and 'allOf'/'anyOf'/'oneOf' is a bit ambiguous. This has been auto-corrected to 'type:object'`,
-              ["PreCheck", "SchemaMissingType"],
-              schema,
-            );
-            schema.type = JsonType.Object;
-            break;
+            // The schema does not have properties or additionalProperties, but it does have allOf/anyOf/oneOf.
```

Comment:
This looks good, my only worry was about the effect on the specs repo.

I am guessing `oneOf` and `anyOf` will not matter because we can't use it right?

But I think there might be cases where people use `allOf` but forgot to specify the type in the base or parent schema.